node-w3capi/index.js

```diff
@@ -18,20 +18,66 @@
 // }
 
 import fs from "fs";
+import ora from "ora";
 import { fetchData } from "./api.js";
-import { renderHTML } from "./render.js";
+import { renderHTML } from "./html.js";
+import { renderMarkdown } from "./markdown.js";
 
 const SOLID_CG_ID = 110151;
 
-const { users, orgs } = await fetchData(SOLID_CG_ID);
+const filenames = {
+  json: {
+    users: "users.json",
+    orgs: "orgs.json"
+  },
+  html: {
+    participants: "participants.html"
+  },
+  md: {
+    affiliations: "affiliations.md"
+  }
+}
 
-fs.writeFileSync("users.json", JSON.stringify(users, null, 2));
-fs.writeFileSync("orgs.json", JSON.stringify(orgs, null, 2));
+const spinner = ora("🐌 Starting").start()
 
-const html = renderHTML(users, orgs, {
+let data
+
+function localDataExists(record) {
+  return Object.values(record).every(filename => fs.existsSync(filename))
+}
+
+// check if exist in the filesystem
+if (localDataExists(filenames.json)) {
+  spinner.succeed("💾 Found existing local data")
+  data = {
+    users: JSON.parse(fs.readFileSync(filenames.json.users, 'utf-8')),
+    orgs: JSON.parse(fs.readFileSync(filenames.json.orgs, 'utf-8'))
+  }
+} else {
+  spinner.start("☕ Fetching data from W3C API")
```

Comment:
This is cute but I miss the useful information that was there in the first place where the simple console.log showed what was being done.

Could that be re-integrated and/or an percentage indicator of the progress?

Reply:
I found previous logs verbose to the point of being useless. For debugging it may be better to actually start a debugger. The only reasonable part was to let the user know that the script is still working since W#C API calls take forever to complete.
Please feel free to adjust log levels to your liking.